lnrpc/routerrpc/router_backend.go

```diff
@@ -603,6 +603,7 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 			payIntent.RouteHints, payReq.RouteHints...,
 		)
 		payIntent.InvoiceFeatures = payReq.Features
+		payIntent.PaymentAddr = payReq.PaymentAddr
 	} else {
 		// Otherwise, If the payment request field was not specified
```

Comment:
We need to take `payment_addr` from the rpc request. Otherwise we can't do key send in combination with single shot mpp as the default payment method.

Reply:
we don't need to use mpp for keysend because intermediaries can't snipe them (since the preimage is in the onion). we could if we want, but idt it adds anything security wise

Reply:
But how do we communicate that we don't want payment address over the rpc if we do a `SendPayment` with the custom key send record? We will default to mpp for 0.9?

Reply:
Discussed offline. We will force old-style payment key send by passing in manual features to sendpayment.

Reminder: if key send gets merged first, the manual feature set needs to be added to `lncli` in this pr.